core/cont/inc/THashList.h

```diff
@@ -36,7 +36,7 @@ class THashTable;
 class THashList : public TList {
 
 protected:
-   THashTable   *fTable;    //Hashtable used for quick lookup of objects
+   THashTable   *fTable;    //! Hashtable used for quick lookup of objects
```

Comment:
Dear Philippe, 

> However I still hesitant as 'requiring' both ClassDef(0) and //! essentially means that the ClassDef(0) is moot/unnecessary ....

Yes, you are right... and you are inspiring me. Thinking a bit more about it, I even get the feeling that the "safest" and most expressive solution would be to use `//!` on the member and Class-Version 1, even though it's technically unnecessary to use a class-version > 0. 

Why? 
The problem is that if some programmer in the future wants to extend e.g. `TSeqCollection` with a new member, it is very likely that he will only add the member and functionality and assume that the member is streamed, since that class is used as baseclass for most other collections. Most of the contributors to ROOT (I believe) are not expert in IO and tend to forget to check / bump the Class-Version on each data-structure change (I also plan to add an incremental check to rootStaticAnalyzer to check for forgotten class-version bumps - we have something like that in our framework-checker). 
So in such a case, there are two ways in which the change can go wrong: 
- The contributor forgets about the class-version, assuming the added member will be streamed. The new functionality will break only after streaming. ROOT will not complain about a forgotten class-version bump, since the realdata-layout has not changed, so this break will be silent. 
- The contributor just bumps the class version after adding the new member, ignoring the special semantics of Class-Version 0. The existing member `fSorted` will now be streamed (even though it is not needed). 

If `TSeqCollection` had class-version 1 and `//!` for the `fSorted` member, I see only one, less harmful way in which things could go wrong: 
- The contributor adds a new member + functionality, but forgets to bump the class-version It will be streamed, nothing will break - and the next time somebody tries to read an old file, ROOT's checksum magic will produce a warning that the class-version bump was forgotten. Still, no real breakage will occur, and the programmer will be alerted. 

That makes me think that Class-Version 0 is not very expressive for any class used directly or indirectly in IO. It's really the single member which is used in a transient way (e.g. to cache something, like the "sorted" state) and not the actual "class" itself. 

So in total, I have the feeling that `//!` on the member and class-version 1 would be better from the safety and readability point of view, even if technically unnecessary. I may be wrong, though, but I'm inspired by the errors I see people doing in our framework code  ;-) . 

What do you think? 

Cheers, 
Oliver